### PR TITLE
Add proper JNI / j9object_t error handling

### PR DESCRIPTION
**Pull request by DanHeidinga**
Signed-off-by: Dan Heidinga <daniel_heidinga@ca.ibm.com>

### CODE REVIEW

File: runtime/jcl/common/sun_reflect_ConstantPool.c [hunk 1]
```diff
@@ -564,35 +564,62 @@ Java_sun_reflect_ConstantPool_getMemberRefInfoAt0(JNIEnv *env, jobject unusedObj
...
 	}
 
 	if ((NULL != classNameObject) && (NULL != nameObject) && (NULL != signatureObject)) {
 		jclass jlString = JCL_CACHE_GET(env, CLS_java_lang_String);
 		jobject array = (*env)->NewObjectArray(env, 3, jlString, NULL);
+		if (array == NULL) {
+			goto done;
+		}
 		(*env)->SetObjectArrayElement(env, array, 0, classNameObject);
+		if ((*env)->ExceptionCheck(env)) {
```
Comment (gacholio) [inline runtime/jcl/common/sun_reflect_ConstantPool.c]: This is a waste - the previous call cannot fail.

File: runtime/jcl/common/sun_reflect_ConstantPool.c [hunk 2]
```diff
@@ -564,35 +564,62 @@ Java_sun_reflect_ConstantPool_getMemberRefInfoAt0(JNIEnv *env, jobject unusedObj
...
+			goto done;
+		}
 		(*env)->SetObjectArrayElement(env, array, 1, nameObject);
+		if ((*env)->ExceptionCheck(env)) {
```
Comment (gacholio) [inline runtime/jcl/common/sun_reflect_ConstantPool.c]: Same here.

File: runtime/jcl/common/sun_reflect_ConstantPool.c [hunk 3]
```diff
@@ -564,35 +564,62 @@ Java_sun_reflect_ConstantPool_getMemberRefInfoAt0(JNIEnv *env, jobject unusedObj
...
+			goto done;
+		}
 		(*env)->SetObjectArrayElement(env, array, 2, signatureObject);
-		(*env)->DeleteLocalRef(env, classNameObject);
-		(*env)->DeleteLocalRef(env, nameObject);
-		(*env)->DeleteLocalRef(env, signatureObject);
-		if (!(*env)->ExceptionCheck(env)) {
-			returnValue = array;
+		if ((*env)->ExceptionCheck(env)) {
```
Comment (gacholio) [inline runtime/jcl/common/sun_reflect_ConstantPool.c]: And here.

File: runtime/jcl/common/sun_reflect_ConstantPool.c (hunk 1 above)
Comment (DanHeidinga) [reply runtime/jcl/common/sun_reflect_ConstantPool.c]: Theory was that jni check would warn on this code without the exception checks

Comment (gacholio) [reply runtime/jcl/common/sun_reflect_ConstantPool.c]: I don't think it will (I suspect we have lots of other places that are not doing the check).

Comment (DanHeidinga) [reply runtime/jcl/common/sun_reflect_ConstantPool.c]: Turns out that jni check will warning for a pending exception when calling `SetObjectArrayElement` but it doesn't set the `jniCheckSetPotentialPendingException` flag

Comment (gacholio) [reply runtime/jcl/common/sun_reflect_ConstantPool.c]: My mistake - it can throw exceptions (though you have to make a programming error to trigger it).

**Comment by gacholio** [comment]
jenkins test sanity plinux jdk11

**Pull request merged**